src/main/java/cn/nukkit/utils/material/BlockType.java

```diff
@@ -0,0 +1,4 @@
+package cn.nukkit.utils.material;
+
+public interface BlockType extends MaterialType {
```

Comment:
Why do the classes in this package exist?

Reply:
At the time when I implemented it, I wasn't really sure where to put it. Didn't want to use blocks package because it contains already a ton of Block implementations, but I think it will be fine now.
Also it is worth noting this PR is not done yet, so some things are still to be changed/cleaned. But the current state should be at least something that already works and can be tested with.